### PR TITLE
Location sorting fix

### LINKED ISSUE #491: Chapters sorting when location is not available
When user's location is not available, the app does the same sorting with location of 0.0, 0.0 coordinates. 
Instead, don't take the location into account when it is not available. Just sort alphabetically.

### PR DESCRIPTION
**Pull request by tasomaniac**
Fixes #491 

Actually there were no problems. Location is null when there is no location. It is not 0,0. 
The problem was the chapter names starting with space char. I've added a deserializer to fix that. 

While doing that, 
Unnecessary comments are removed.
Location is given in constructor to the ChapterComparator. No need to use App class in the comparator.

### CODE REVIEW

**Comment by friedger** [comment]
Not sure whether we should fix the chapters name, where do we stop fixing
issues? Preferably this is fixed in the hub!

**Comment by tasomaniac** [comment]
Yes it should be done in hub. But Frisbee is in much better shape than hub. Hub has lots of issues. And I already implemented it :)

File: app/src/main/java/org/gdg/frisbee/android/api/model/ChapterDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Type;
+
+public class ChapterDeserializer implements JsonDeserializer<Chapter> {
+    @Override
+    public Chapter deserialize(JsonElement jsonElement, Type type, JsonDeserializationContext jsonDeserializationContext) throws JsonParseException {
+        Chapter chapter = new Gson().fromJson(jsonElement, type);
+        chapter.name = chapter.name.trim();
```
Comment (friedger) [inline app/src/main/java/org/gdg/frisbee/android/api/model/ChapterDeserializer.java]: Too fast merged, it is about having two spaces between 'GDG'  and city name. Not about spaces that can be trimmed.

**Comment by friedger** [comment]
Chapter.getShortname needs fixing

File: app/src/main/java/org/gdg/frisbee/android/api/model/ChapterDeserializer.java (hunk 1 above)
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/api/model/ChapterDeserializer.java]: Do we have 'GDG'  keyword coming from the API for every chapter? In my manual tests, it seems that this PR fixes the problem described in the issue.

Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/api/model/ChapterDeserializer.java]: I didn't quite follow. If you can describe the problem I can fix it :)

Comment (friedger) [inline app/src/main/java/org/gdg/frisbee/android/api/model/ChapterDeserializer.java]: On Gitter

**Pull request merged**